### PR TITLE
Refactor list interface to have iterator

### PR DESCRIPTION
**Pull request by greenmonn**
Refactor list interface to have Iterator subclass.

Insert and Erase element on any position using iterator.

### CODE REVIEW

File: LoRaWan/List.h [hunk 1]
```diff
@@ -6,111 +6,102 @@
...
-
-    T *GetFirst();
-
-    T *GetLast();
-
-    class iterator {
-    private:
+    class Iterator {
+    public:
         Item *item;
```
Comment (greenmonn) [inline LoRaWan/List.h:39]: Item이 public으로 드러나는 게 좋아 보이지 않네요..

File: LoRaWan/List.h [hunk 2]
```diff
@@ -6,111 +6,102 @@
...
-T *List<T>::GetFirst() {
-    if (this->first == NULL) {
-        return NULL;
-    }
+typename List<T>::Iterator List<T>::Insert(const List<T>::Iterator &iter, T& data) {
+    Item *newItem = new Item(data);
 
-    return this->first->data;
+    return (insertItem(newItem, iter.item));
 }
```
Comment (goofcode) [inline LoRaWan/List.h:98]: missing call of Iterator constructor

File: LoRaWan/List.h [hunk 3]
```diff
@@ -6,111 +6,102 @@
...
-
-    void Remove(T *data);
-
-    T *GetFirst();
-
-    T *GetLast();
-
-    class iterator {
+    class Iterator {
+        friend class List;
```
Comment (greenmonn) [inline LoRaWan/List.h:37]: `Friend`!!

Comment (goofcode) [reply LoRaWan/List.h:37]: **gooood**

**Review by goofcode — APPROVED** [review]

**Pull request merged**